ui/src/pages/login/Login.js

```diff
@@ -2,14 +2,59 @@
  * Copyright (c) 2019, Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
  * SPDX-License-Identifier: Apache-2.0
  */
-import React, { useState } from "react";
-import { Grid, CircularProgress, Typography, Button, TextField, Fade, Select, MenuItem, InputLabel, FormControl } from "@material-ui/core";
+import React, { useState , useEffect} from "react";
+import { Grid, CircularProgress, Typography, Button, Fade, Select, MenuItem, InputLabel, FormControl } from "@material-ui/core";
 import { withRouter } from "react-router-dom";
 import useStyles from "./styles";
 import logo from "./logo.svg";
-import { useUserDispatch, loginUser, redirectToDablLoginPage } from "../../context/UserContext";
-import { isLocalDev, handlePartiesLoad, ledgerId } from "../../config";
-import { DablPartiesInput } from '@daml/dabl-react'
+import rp from "request-promise";
+import { useUserDispatch, loginUser } from "../../context/UserContext";
+import { createTokenAll, httpBaseUrl, isLocalDev } from "../../config";
+import { addPath, getDablPartyParticipants } from "../../components/Util";
+import participants from '../../participants.json';
+
+// getDisplayPartyNames
+export function useSortedPartyNames() {
+  var [parties, setParties] = useState([]);
+  useEffect(() => {
+    getPartiesI().then(ps => setParties((ps)));
```

Comment:
untested, but I think you wanted sth like this:
```suggestion
    getPartiesI().then(ps => setParties(getParties(ps)));
```

Reply:
(you can also inline the `getParties`)

Reply:
that just sorts them doesnt it? why is that required?

Reply:
yes, just sorts. a blurred memory: in some dropdown list we wanted the parties alphabetically sorted (someone wants to select a Supplier / Buyer / ... as a party, or on the login page.

i have no idea if you need them sorted here, but since you included the building blocks, they could be used (or removed)